src/sentry/lang/native/plugin.py

```diff
@@ -152,6 +179,92 @@ def preprocess_step(self, processing_task):
         if options.get('symbolserver.enabled'):
             self.fetch_system_symbols(processing_task)
 
+        if self.use_symbolicator:
+            self.run_symbolicator(processing_task)
+
+    def run_symbolicator(self, processing_task):
+        request_id_cache_key = request_id_cache_key_for_event(self.data)
+
+        # TODO(markus): Make this work with minidumps. An unprocessed minidump
+        # event will not contain unsymbolicated frames, because the minidump
+        # upload already happened in store.
+
+        images = list(get_path(self.data, 'debug_meta', 'images', filter=True))
+        if not images:
+            return
+
+        stacktraces = []
+        processable_frames = []
+        for stacktrace_info, pf_list in processing_task.iter_processable_stacktraces():
+            registers = stacktrace_info.stacktrace.get('registers') or {}
+
+            pf_list = [
+                # This condition is copied from iter_processable_frames
+                pf for pf in reversed(pf_list)
+                if pf.processor == self
+            ]
+
+            frames = [
+                {'instruction_addr': pf['instruction_addr'],
+                 'trust': pf.get('trust')}
+                for pf in pf_list
+            ]
+
+            stacktraces.append({
+                'registers': registers,
+                'frames': frames
+            })
+
+            processable_frames.append(pf_list)
+
+        rv = run_symbolicator(stacktraces=stacktraces, modules=images,
+                              project=self.project, arch=self.arch,
+                              signal=self.signal,
+                              request_id_cache_key=request_id_cache_key)
+        if not rv:
+            return
+
+        # TODO(markus): Set signal and os context from symbolicator response,
+        # for minidumps
+
+        assert len(images) == len(rv['modules']), (images, rv)
```

Comment:
For future reference: Depending on how minidumps are processed, this list might not have been filled at that point.

Reply:
@jan-auer should we just write this as code comment? This seems like a suboptimal way to persist this information.

Reply:
As you wish, we're forced to revisit this with the very next follow-up.

Reply:
sure, it's just that this function might need to be rewritten anyway since it `zip`s over existing images